docs/ppl-zkp-design.md

```diff
@@ -0,0 +1,125 @@
+
+- WalletID
+- SleeveID
+- IOUType
+- UOM (Unit of Measurement)
+- PublicKey
+- NotaryID (the primary notary for this wallet)
+
+(Note: we could replace PublicKey with a list, allowing for multiple owners with 1-of-N or k-of-N semantics. For simplicity, this functionality not described here, but the changes for this would be minimal.)
+
+A sleeve stores fungible IOUs (of one type) for one owner. There is one primary notary associated with this sleeve. This notary is responsible for preventing double spends and generally verifying the integrity of the sleeve and associated transctions.
+
+### State
+
+A state consists of:
+
+- StateID (unique across the system)
+- SleeveID
+- Amount
+- StateHash
+
+The StateHash is simply a hash of the StateID+SleeveID+Amount. 
+
+A state represents an amount of `UOM` that has been deposited into a sleeve. A state can be _active_ or _dropped_. An active state is a UTXO—indicating that the amount represented by it has not yet been used in another transaction, so is a valid candidate to be an input for a future transaction. A dropped state is one which has been used in a transaction, and hence cannot be used again.
+
+Any transaction consists of taking one or more active states, dropping them, and redistributing the total UOM from the input states into one or more newly created output states which will now be active states.
+
+In the rest of this document we use the words _spent_ and _unspent_ as synonyms for active and dropped states, because it makes the examples easier to understand, but keep in mind that a sleeve can hold non-currency values in which case spent and unspent words would be misleading.
+
+
+### Simple Transaction
+
+A simple transaction is one in which all the input states belong to the same owner (called the _sender_ in this example) and the primary notary for all of them is the same. The output states can belong to different owners (referred to as _receivers_).
+
+We will go through an example transaction with two inputs and two outputs. 
+
+#### Sender steps
+
+In this case, the owner creates a transaction core data structure as follows:
+
+- Input states
+  - SIn1: SIn1ID, SIn1SleeveID, SIn1Amount, SIn1Hash
+  - SIn2: SIn2ID, SIn2SleeveID, SIn2Amount, SIn2Hash
+- Output states
+  - SOut1: SOut1ID, SOut1SleeveID, SOut1Amount, SOut1Hash
+  - SOut2: SOut2ID, SOut2SleeveID, SOut2Amount, SOut2Hash
+- TxCoreHash = Hash(SIn1Hash + SIn2Hash + SOut1Hash + SOut2Hash)
+
+Note: this transaction core data structure is known and visible only to the owner. At no point are the SleeveIDs and Amounts of any of the states revealed to any of the notaries or on the public chain. The details of SOut1 and SOut2 will be sent to the receivers by the sender using some other communication method.
+
+Based on this, the sender creates a transaction packet consisting of
+this information:
+
+- TxCoreHash
+- SIn1ID, SIn1Hash, TxCoreHash, SIn1Proof
+- SIn1ID, SIn1Hash, TxCoreHash, SIn1Proof
+- SOut1ID, SOut1Hash
```

Comment:
I suspect all attributes for out states that need to be recorded in the public ledger will be required here.